### PR TITLE
Team showcase advertisement on profile page

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers 

For desktop and mobile:

* On the profile view, if you are in at least one team but have no teams showcased, show an invitation to showcase

* When clicked, show a new popup dialog showing each team you're in and its showcasing status -- whether it's promoted, and if not whether you can promote it

* When clicked, change the promoted status of that team

* Use waiting spinners on the button and while loading

### CODE REVIEW

File: shared/profile/showcase-team-offer/index.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  Text,
+} from '../../common-adapters'
+import {teamWaitingKey} from '../../constants/teams'
+
+import type {RowProps, Props} from './index'
+
+const TeamRow = ({canShowcase, name, isOpen, membercount, onPromote, showcased, waiting}: RowProps) => (
+  <Box style={globalStyles.flexBoxColumn}>
+    <Box
+      key={name}
```
Comment (chrisnojima) [inline shared/profile/showcase-team-offer/index.js]: key seems to serve no purpose here

File: shared/profile/showcase-team-offer/index.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        <Box style={globalStyles.flexBoxRow}>
+          <Text type="BodySemibold">{name}</Text>
+          {isOpen && <Meta title="OPEN" style={styleMeta} />}
+        </Box>
+        <Box style={{...globalStyles.flexBoxRow, alignItems: 'center'}}>
+          <Text type="BodySmall">{membercount + ' member' + (membercount !== 1 ? 's' : '')}</Text>
+        </Box>
+      </Box>
+      {showcased ? (
+        <Button
```
Comment (chrisnojima) [inline shared/profile/showcase-team-offer/index.js]: in this case we should likely just have one button and use showcased to switch the label / onClick param etc

File: shared/profile/showcase-team-offer/index.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+const styleMeta = {
+  alignSelf: 'center',
+  backgroundColor: globalColors.green,
+  borderRadius: 1,
+  marginLeft: globalMargins.xtiny,
+  marginTop: 2,
+}
+
+const PopupWrapped = (props: Props) => (
```
Comment (chrisnojima) [inline shared/profile/showcase-team-offer/index.js]: @akalin another example of crappy stuff we currently do to handle mobile/desktop popup stuff

File: shared/teams/container.js [hunk 4]
```diff
@@ -47,17 +48,7 @@ const mapDispatchToProps = (dispatch: Dispatch) => ({
...
-    const bName = b.toUpperCase()
-    if (aName < bName) {
-      return -1
-    } else if (aName > bName) {
-      return 1
-    } else {
-      return 0
-    }
-  })
+  teamnames.sort((a, b) => sortTeamnames(a, b))
```
Comment (chrisnojima) [inline shared/teams/container.js]: could just be `teamnames.sort(sortTeamnames)`

**Comment by cjb** [comment]
@chrisnojima Did these, also split out a new action for getting publicity settings to avoid having to do a full teamGet.  Ready for another look, but I won't merge since we're in freeze.

File: shared/profile/index.desktop.js [hunk 5]
```diff
@@ -343,31 +343,52 @@ class ProfileRender extends PureComponent<Props, State> {
...
               </Box>
               <Box style={styleProofs}>
-                {!loading &&
-                  this.props.userInfo.showcasedTeams.length > 0 && (
-                    <Box style={{...globalStyles.flexBoxColumn, paddingBottom: globalMargins.small}}>
-                      <Box style={globalStyles.flexBoxRow}>
-                        <Text type="BodySmallSemibold">Teams:</Text>
+                {!loading && (
+                  <Box style={{...globalStyles.flexBoxColumn, paddingBottom: globalMargins.small}}>
+                    {(this.props.userInfo.showcasedTeams.length > 0 ||
```
Comment (chrisnojima) [inline shared/profile/index.desktop.js]: Profile code is pretty old/messy. We don't need to make this change now but in future stuff I'd advocate simpler logic in jsx and breaking up the components into sub components
```
const showEdit = (this.props.userInfo.showcasedTeams.length > 0 || (this.props.isYou && this.props.youAreInTeams)
```

```
{showEdit && <EditControl />}
```

Comment (chrisnojima) [reply shared/profile/index.desktop.js]: Could even pull these sub sections into their own components
```
{showEdit && <EditControl />}
```

Comment (cjb) [reply shared/profile/index.desktop.js]: Cool, makes sense.  I'll merge this and open a new PR to clean it up.

Comment (chrisnojima) [reply shared/profile/index.desktop.js]: lets just ticket and leave in the backlog. i think we'll want to revisit this and merge desktop and native also

**Pull request merged**